### PR TITLE
tests: improve cache tests assertions

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
tests

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**
no
**Summary**
 improve cache tests assertions

**Does this PR introduce a breaking change?**
NO

**Other information**
NO

### CODE REVIEW

File: test/utils/test-utils.js [hunk 1]
```diff
@@ -248,6 +248,10 @@ const runInfo = (args, testPath) => {
     return run(testPath, ['info'].concat(args), false);
 };
 
+const windowsPathConverter = (path) => {
+    return path.replace(/\\/g, '\\\\');
```
Comment (alexander-akait) [inline test/utils/test-utils.js:252]: Never do there, you have `path.normalize`, replacing `\` on `//` is not a safe in some cases

File: test/core-flags/cache-flags.test.js [hunk 2]
```diff
@@ -211,6 +254,11 @@ describe('cache related flags from core', () => {
         expect(stderr).toContain("Compilation 'compiler-cache' finished");
         expect(stdout).toContain("type: 'filesystem'");
         expect(stdout).toContain('buildDependencies');
+        if (isWindows) {
+            expect(stdout).toContain('cache-core-flag-test-build-dependencies-in-config');
+        } else {
+            expect(stdout).toContain(`'${cacheLocation}'`);
+        }
```
Comment (alexander-akait) [inline test/core-flags/cache-flags.test.js:261]: Avoid this, you can `normalize` path

**Comment by snitin315** [comment]
No luck with `path.normalize()` 😞

**Comment by snitin315** [comment]
Yes!

**Pull request closed without merging**